### PR TITLE
Fix an inconsistency between the mute status icon in full screen mode…

### PR DESCRIPTION
**Pull request by shahinyan11**
### Fixed Issues

$ https://github.com/Expensify/App/issues/38307
PROPOSAL: https://github.com/Expensify/App/issues/38307#issuecomment-1998334705

### Tests

1. Send a video as an attachment inside of a chat
2. Open the video in full screen mode
3. Mute the video
4. Go out of full screen mode and verify that the sound icon is muted

- [x] Verify that no errors appear in the JS console

### Offline tests

N/A

### QA Steps

Same as in the Tests section.
- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/46921547/00a7f375-6c5b-4187-bf6d-5ab44e7dbbfb

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/46921547/3aaff422-e46f-4cfc-9ab5-7c24bda9b8fa

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/46921547/97787805-d055-411e-9e77-fc1d5c50394e

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/46921547/4db8dc43-fa79-451d-b5ee-ebacfb74eebb

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/46921547/df514384-2bfc-4222-bc32-e0efbc7a5842

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/46921547/5bf3c37d-77fe-4161-a398-ad9daab5288a

</details>

### CODE REVIEW

File: src/components/VideoPlayer/BaseVideoPlayer.js [hunk 1]
```diff
@@ -134,6 +136,15 @@ function BaseVideoPlayer({
     const handleFullscreenUpdate = useCallback(
         (e) => {
             onFullscreenUpdate(e);
+
+            if (e.fullscreenUpdate === VideoFullscreenUpdate.PLAYER_DID_DISMISS) {
+                currentVideoPlayerRef.current.getStatusAsync().then((status) => {
```
Comment (jayeshmangwani) [inline src/components/VideoPlayer/BaseVideoPlayer.js:141]: @shahinyan11 Lint is failing, please fix lint warnings

**Comment by shahinyan11** [comment]
@jayeshmangwani I noticed things that I want to warn you about
1. The volume icon always displayed as muted in full screen mode on IOS platforms ( `native` and `mWeb`). but when pressed it switches the sound, the icon just doesn’t update. And it's a package error
2. On android `native`, the native control does not contain volume icon in full screen mode

File: src/components/VideoPlayer/BaseVideoPlayer.js [hunk 2]
```diff
@@ -134,6 +136,17 @@ function BaseVideoPlayer({
...
+                    if (status.volume > 0 && !status.isMuted) {
+                        return;
+                    }
+
+                    updateVolume(0);
```
Comment (jayeshmangwani) [inline src/components/VideoPlayer/BaseVideoPlayer.js:146]: @shahinyan11 If we set the updateVolume to status.volume, the volume value can be synced between fullscreen and mini player, right? like below.
```diff
+if(status.value === 0 || status.isMuted) {
+     updateVolume(0);
+     return;
+}
+     updateVolume(status.volume);
```

**Comment by jayeshmangwani** [comment]
@shahinyan11 And we must fix the issue below in this PR;

1. Open a video to upload in mini player
2. Mute the Video
3. Play Video
4. Open In fullscreen
5. Unmute the video using the slider or press the volume icon
6. Press the Close fullscreen
7. On mini player video should be unmute

https://github.com/Expensify/App/assets/35371050/c2f6fec2-9156-4dd2-ad66-3775874340d3

File: src/components/VideoPlayer/BaseVideoPlayer.js (hunk 2 above)
Comment (shahinyan11) [reply src/components/VideoPlayer/BaseVideoPlayer.js:146]: What about `updateVolume(status.value === 0 || status.isMuted ? 0 : status.volume);` ?

**Comment by shahinyan11** [comment]
@jayeshmangwani Video you shared on simulator or real device? I'm testing on the iOS simulator and the volume icon does not change at all when I click on it. 

https://github.com/Expensify/App/assets/46921547/c46b1709-2af1-4a67-b941-90bfe43e1069

**Comment by jayeshmangwani** [comment]
@shahinyan11 Simulator `iPhone SE - ios version 16.4`

**Comment by jayeshmangwani** [comment]
16.4 is deprecated or unsupported?

**Comment by shahinyan11** [comment]
@jayeshmangwani I tried it on 16.4 and it works like in your video . But it is a package bug. The updating volume in full screen mot does not call any event

**Comment by jayeshmangwani** [comment]
>  I tried it on 16.4 and it works like in your video . But it is a package bug. The updating volume in full screen mot does not call any event. I even tried to update package to latest version, but it did not help

I will test more cases tomorrow and update it here.

**Comment by shahinyan11** [comment]
@jayeshmangwani Don't have any updates?

**Comment by jayeshmangwani** [comment]
@shahinyan11 Not yet, and I was not able to find any issue where it mentioned that 16.4 will not trigger the volume call

**Comment by jayeshmangwani** [comment]
@thienlnam I was able to reproduce this [issue](https://github.com/Expensify/App/pull/39479#issuecomment-2037251863) in the version iOS 16.4. Should we try to solve this issue here? As @shahinyan11 mentioned here, it's a package bug https://github.com/Expensify/App/pull/39479#issuecomment-2037976899. And also @shahinyan11 Can you link any issue or discussion that confirms this is package issue?

**Comment by jayeshmangwani** [comment]
Thanks for the update; I am testing this PR on desktop and mWeb Android right now.

**Comment by shahinyan11** [comment]
@jayeshmangwani Could you test [this](https://github.com/Expensify/App/pull/39479#issuecomment-2037251863) bug and confirm that it is fixed ?

**Comment by jayeshmangwani** [comment]
> @jayeshmangwani Could you test [this](https://github.com/Expensify/App/pull/39479#issuecomment-2037251863) bug and confirm that it is fixed ?

@shahinyan11 That issue is fixed, but I am facing a problem that is not looking good.

1. Open a video in mini player > audio is mute 
2. Start the video and open it in a fullscreen player
3. Press unmute and close the fullscreen player
4. Now observe the mini-player play/pause button; it is flickering from play > pause > play

Please check the video below, especially 5-6 seconds timestamp 


https://github.com/Expensify/App/assets/35371050/1e09ea0d-13ff-49ee-98c0-4f9ec5bc49e5

**Comment by shahinyan11** [comment]
@jayeshmangwani I updated my change to decrease re-rendering count of play/pause icon and it now looks better but the bug is not fixed completely. But If we still want to fix it  completely, I think it should be done around different issue.

**Comment by jayeshmangwani** [comment]
> @jayeshmangwani I updated my change to decrease re-rendering count of play/pause icon and it now looks better but the bug is not fixed completely. But If we still want to fix it completely, I think it should be done around different issue.

@shahinyan11 I will re-test this PR today, My iOS build was messed up from yesterday, fixed now.

File: src/components/VideoPlayer/BaseVideoPlayer.tsx [hunk 3]
```diff
@@ -145,6 +147,15 @@ function BaseVideoPlayer({
 
             if (event.fullscreenUpdate === VideoFullscreenUpdate.PLAYER_DID_DISMISS) {
                 isFullScreenRef.current = false;
+
+                currentVideoPlayerRef.current?.getStatusAsync?.().then((status) => {
+                    if (!('isMuted' in status)) {
```
Comment (jayeshmangwani) [inline src/components/VideoPlayer/BaseVideoPlayer.tsx:152]: Please add a comment here as to why we added this line of code.

File: src/components/VideoPlayer/BaseVideoPlayer.tsx [hunk 4]
```diff
@@ -145,6 +147,15 @@ function BaseVideoPlayer({
...
+                        return;
+                    }
+
+                    updateVolume(status.isMuted ? 0 : status.volume || 1);
```
Comment (jayeshmangwani) [inline src/components/VideoPlayer/BaseVideoPlayer.tsx:156]: here `|| 1` is necessary ?

**Comment by jayeshmangwani** [comment]
@shahinyan11 I tested the latest code on desktop, web, iOS, and mWeb Safari, and the results seem good to me; I have suggested a few changes.

File: src/components/VideoPlayer/BaseVideoPlayer.tsx (hunk 4 above)
Comment (jayeshmangwani) [reply src/components/VideoPlayer/BaseVideoPlayer.tsx:156]: Instead, can we just set the volume like this: `updateVolume(status.isMuted ? 0: status.volume)`?

Comment (shahinyan11) [reply src/components/VideoPlayer/BaseVideoPlayer.tsx:156]: No if we remove the `|| 1` we will have the [bug](https://github.com/Expensify/App/pull/39479#issuecomment-2037251863) mentioned above if we turn on the video sound in the `5th` step by pressing the volume icon.

File: src/components/VideoPlayer/BaseVideoPlayer.tsx (hunk 3 above)
Comment (jayeshmangwani) [reply src/components/VideoPlayer/BaseVideoPlayer.tsx:153]: @shahinyan11 The comment looks good, but please update the comment to `Synchronize the volume settings when switching from full screen to mini-player mode.` We can later confirm with the internal engineer.

Comment (jayeshmangwani) [reply src/components/VideoPlayer/BaseVideoPlayer.tsx:153]: And why are we returning early if the `isMuted` is not in the `status`?

Comment (shahinyan11) [reply src/components/VideoPlayer/BaseVideoPlayer.tsx:153]: For hiding Typescript error

**Comment by jayeshmangwani** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I verified tests pass on **all platforms** & I tested again on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/Expensify/App/assets/35371050/233cb372-12c0-4a8e-b0fe-e24bba967cd1




</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/Expensify/App/assets/35371050/1e40d748-c979-4e45-9e9d-b67d7077429d




</details>

<details>
<summary>iOS: Native</summary>

https://github.com/Expensify/App/assets/35371050/e29db28b-4a08-4131-824e-2b815e9c4aad




</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/Expensify/App/assets/35371050/01a06e95-9993-4d91-b5e3-04cc03ddd8dc




</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/Expensify/App/assets/35371050/8853d6b5-c2c7-442d-a8c2-6774f75285f3




</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/Expensify/App/assets/35371050/ccdebd2b-e5b0-4a35-8a7a-effa7eef686b




</details>

**Review by jayeshmangwani — APPROVED** [review]
Changes LGTM 🚀

**Review by thienlnam — APPROVED** [review]

**Pull request merged**